reference/1_application_flow/1_2_iterators/README.md

```diff
@@ -0,0 +1,167 @@
+# Iterators / loops
+
+The closes analogy to an imperative "foreach" statement in Cranq is an "Iterator". It comes in two flavours:
+- ```data/array/Iterator``` to work with arrays
+- ```data/dictionary/Iterator``` for dictionaries/key-value pairs
+
+As it's name suggests, an iterator iterates through a data collection, and emits a signal with each element on it's output. 
+
+However, the iterator creates a new signal with distinct tags for each element - as a result:
+- It is inherently parallel
+- The order in which the outputs are emitted, is not deterministic
+- The output signals of an iterator will need to be synchronized, if needed
+
+## Example - simple foreach
+
+> **_Try out:_**
+>
+> Place node **tutorials/app_flow/iterator_array** or **[download .cranqj](cranqj/iterator_array.cranqj)** 
+
+Consider a simple application, that takes an array as it's input, and outputs each element.
+
+![](images/2021-07-09-12-34-21.png)
+
+- The example uses a store as data source:
+  ```json
+    [1,8,13,7,2,10,84,26]
+  ```
+- Connect this store to an ```data/array/Iterator```
+
+Take note of the output of this application:
+```
+1 @start:0
+8 @start:1
+13 @start:2
+7 @start:3
+2 @start:4
+10 @start:5
+84 @start:6
+26 @start:7
+```
+The logger being used here outputs the signal data, along with it's tag. 
+If you are unfamiliar with tags, please have a look at this[TODO: where?]
+
+Observing the tags here, we see that the iterator forked our original signal for each array item.
+
+Let's extend this example with a ```flow/Delayer```, representing an expensive processing operation - this will block the flow of data for each element, for a specified amount of time:
+
+![](images/2021-08-03-09-19-45.png)
+
+Upon executing this code, the parallel nature of the processing can be observed.

```

Comment:
I think the Delayer should be random to truly demonstrate the parallel nature of the processing. With the same amount of  delay the operation will keep the inputs' order and the parallel processing won't be visible to the user.

Reply:
Adding clarification